### PR TITLE
Migrate the Activity and NavGraph to Compose

### PR DESCRIPTION
**Pull request by manuelvicnt**

### CODE REVIEW

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoNavGraph.kt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import com.example.android.architecture.blueprints.todoapp.addedittask.AddEditTaskScreen
+import com.example.android.architecture.blueprints.todoapp.statistics.StatisticsScreen
+import com.example.android.architecture.blueprints.todoapp.taskdetail.TaskDetailScreen
+import com.example.android.architecture.blueprints.todoapp.tasks.TasksScreen
+import com.example.android.architecture.blueprints.todoapp.util.AppModalDrawer
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@Composable
+fun JetnewsNavGraph(
```
Comment (tunjid) [inline app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoNavGraph.kt:47]: The file name doesn't seem to match the function name

Comment (tunjid) [inline app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoNavGraph.kt:47]: The file name doesn't seem to match the function name

**Comment by JoseAlcerreca** [comment]
Let's fix tests first
 - enable CI
 - Rename TasksActivityTests - > TasksTests
 - Remove IR and waitForIdle from TasksActivityTests
 - Start ComponentActivity instead of TasksActivty
 - Create a private setContent function, like in other tests and call it in each test, after the repo setup

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/TasksActivity.kt [hunk 2]
```diff
@@ -13,17 +13,24 @@
...
-import com.example.android.architecture.blueprints.todoapp.ViewModelFactory
-
-fun Fragment.getViewModelFactory(): ViewModelFactory {
-    val repository = (requireContext().applicationContext as TodoApplication).taskRepository
-    return ViewModelFactory(repository, this)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            AppCompatTheme {
+                JetnewsNavGraph()
```
[the remark this answers is not in the document]
Comment (manuelvicnt) [reply app/src/main/java/com/example/android/architecture/blueprints/todoapp/TasksActivity.kt]: Good catch! Fixed thanks

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailState.kt [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.LifecycleOwner
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+@Composable
+fun rememberTaskDetailState(
```
Comment (JoseAlcerreca) [inline app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailState.kt:35]: Same, comments

**Review by JoseAlcerreca — APPROVED** [review]

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailState.kt (hunk 3 above)
Comment (manuelvicnt) [reply app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailState.kt:35]: Done! Thanks

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskState.kt [hunk 4]
```diff
@@ -41,7 +42,7 @@ fun rememberAddEditTaskState(
     coroutineScope: CoroutineScope = rememberCoroutineScope()
 ): AddEditTaskState {
     val currentOnTaskUpdateState by rememberUpdatedState(onTaskUpdate)
-    return remember(scaffoldState, viewModel, taskId, lifecycleOwner, context, coroutineScope) {
+    return remember(taskId, viewModel, scaffoldState, lifecycleOwner, context, coroutineScope) {
```
[the remark this answers is not in the document]
Comment (manuelvicnt) [reply app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskState.kt:50]: Done, thanks!

File: app/src/androidTestMock/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksActivityTest.kt [hunk 5]
```diff
@@ -105,188 +74,209 @@ class TasksActivityTest {
...
     fun unregisterIdlingResource() {
         IdlingRegistry.getInstance().unregister(EspressoIdlingResource.countingIdlingResource)
-        IdlingRegistry.getInstance().unregister(dataBindingIdlingResource)
+    }
+
+    @Before
+    fun init() {
+        // Run on UI thread to make sure the same instance of the SL is used.
+        runOnUiThread {
+            ServiceLocator.createDataBase(getApplicationContext(), inMemory = true)
```
[the remark this answers is not in the document]
Comment (manuelvicnt) [reply app/src/androidTestMock/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksActivityTest.kt:72]: As discussed offline, leaving it as an e2e test, thanks!

**Pull request merged**